### PR TITLE
Fixed Defect in Test Case for Glue:R0036_0.

### PR DESCRIPTION
**Pull request by ben-Draeger**
Fixed Defect in Test Case for Glue:R0036_0.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Adherence to coding conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Adherence to javadoc conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* config update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* SDCcc executable ran against a test device (if necessary)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

**Review by maximilianpilz — CHANGES_REQUESTED** [review]
please make use of the new license check on the main branch and update the copyright notices in the files that you changed

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - autocommit of the database being on, it is now always off
 - local address resolver sometimes giving a wrong address
 - local address resolver doing a probe which may not be tolerated by some peers
+- Test Case for Glue:R0036_0 now accepts SOAPFault as a valid answer for Subscribe messages.
```
Comment (maximilianpilz) [inline CHANGELOG.md:20]:
```suggestion
- the test case for Glue:R0036_0 not accepting a SOAPFault as a valid answer for Subscribe messages
```

**Review by maximilianpilz — CHANGES_REQUESTED** [review]

File: sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/direct/DirectSubscriptionHandlingTest.java [hunk 2]
```diff
@@ -596,7 +596,14 @@ public void onNotification(final NotificationObject message) {
             reportTestData.setSubscription(result);
             reportTestData.setEventSink(eventSink);
         } catch (InterruptedException | ExecutionException e) {
-            fail("encountered exception while subscribing to " + reportTestData.getReportName(), e);
+            if (e instanceof ExecutionException && e.getCause() instanceof SoapFaultException) {
+                // according to WS-Eventing, answering a Subscribe with a SOAPFault is a normal way of
+                //   declining a subscription. We hence interpret it as "Report is not supported" and do
+                //   not fail the test case.
+                result = null;
```
Comment (maximilianpilz) [inline sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/direct/DirectSubscriptionHandlingTest.java:603]: add a log message here

Comment (ben-Draeger) [reply sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/direct/DirectSubscriptionHandlingTest.java]: Done.

**Review by maximilianpilz — APPROVED** [review]

**Pull request merged**